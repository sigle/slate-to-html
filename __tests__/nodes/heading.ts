import { generateHtml } from '../../src';

it('render heading', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'heading-one',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Title 1',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(generateHtml(slateJSON)).toMatchSnapshot();
});

it('render heading 2', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'heading-two',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Title 2',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(generateHtml(slateJSON)).toMatchSnapshot();
});

it('render heading 3', () => {
  const slateJSON = {
    object: 'value',
    document: {
      object: 'document',
      data: {},
      nodes: [
        {
          object: 'block',
          type: 'heading-three',
          data: {},
          nodes: [
            {
              object: 'text',
              text: 'Title 3',
              marks: [],
            },
          ],
        },
      ],
    },
  };
  expect(generateHtml(slateJSON)).toMatchSnapshot();
});
