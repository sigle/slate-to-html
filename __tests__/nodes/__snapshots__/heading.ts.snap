// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render heading 1`] = `"<h2>Title 1</h2>"`;

exports[`render heading 2 1`] = `"<h3>Title 2</h3>"`;

exports[`render heading 3 1`] = `"<h3>Title 3</h3>"`;
